cmd/kubectl-gadget/utils/trace.go

```diff
@@ -711,3 +711,37 @@ func genericStreamsDisplay(
 		}
 	}
 }
+
+// DeleteTraceByGadgetName removes all traces with this gadget name
+func DeleteTracesByGadgetName(gadget string) error {
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return err
+	}
+
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("gadgetName=%s", gadget),
+		FieldSelector: fields.Everything().String(),
+	}
+	return traceRestClient.
+		Delete().
+		Namespace("gadget").
+		Resource("traces").
+		VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+		Do(context.TODO()).
+		Error()
+}
+
+func ListTracesByGadgetName(gadget string) ([]gadgetv1alpha1.Trace, error) {
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("gadgetName=%s", gadget),
+		FieldSelector: fields.Everything().String(),
```

Comment:
Is it possible to do this with a field selector instead of using the label?

```
FieldSelector: fmt.Sprintf("spec.gadget=%s", gadget)
```
We have an example of that, but for another resource:
https://github.com/kinvolk/inspektor-gadget/blob/f5e0b23afbf4659c0ed8a4b60d7fc63d6c7ebac1/cmd/kubectl-gadget/utils/exec.go#L50

Reply:
It didn't work: `field label not supported: spec.gadget`. Kubernetes objects have a [`ToSelectableFields`](https://github.com/kubernetes/kubernetes/blob/c2882f314218af489267b7163b2ca09c79ffb00a/pkg/registry/core/pod/strategy.go#L306) function to set the fields that could be used in the lookup. Do you have any idea where can be implement them for the trace resource?

Reply:
It seems that for custom resources, the field selector can only work with the namespace and name but not arbitrary fields.

https://github.com/kubernetes/kubernetes/issues/79233
https://github.com/kubernetes/kubernetes/issues/53459

Reply:
Thanks for the clarification!